wireless_modules/wireless_module.py

```diff
@@ -0,0 +1,100 @@
+"""
+A class structure to read and collate data from different sensors into a dictionary and send through MQTT
+"""
+import ujson
+import utime
+from mqtt_client import Client
+
+
+class WirelessModule:
+    def __init__(self, client_id, broker):
+        """
+        Initialises the wireless module
+        :param client_id: A string representing the mqtt client id
+        :param broker: A string representing the IP address or web domain of the mqtt broker
+        """
+        self.sensors = None
```

Comment:
Actually we can instead have an `exception` raised if `_read_sensors` is called without first setting the `sensors`. Since the `_read_sensors` is only called in the `run` method and once we enter that method there's no way of setting the sensors so no point running that method at all. Let me know what you think.

Reply:
I don't think there's any reason we should prevent the wireless module from running if no sensors are added. If we allow no sensors to be added, it lets people test the code on their ESP32 even if they don't have any sensors to connect to it (or they can't be bothered to connect them).